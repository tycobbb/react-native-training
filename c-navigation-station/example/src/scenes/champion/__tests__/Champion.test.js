/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Champion } from '../Champion'

// setup
let subject

function loadSubject (props) {
  subject = shallow(<Champion {...props} />)
}

// spec
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('renders properly', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})
