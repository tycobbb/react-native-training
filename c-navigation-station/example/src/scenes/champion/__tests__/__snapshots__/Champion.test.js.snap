// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Champion />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Champion
      </Text>,
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Champion",
        "disabled": false,
        "ellipsizeMode": "tail",
      },
      "ref": null,
      "rendered": "Champion",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
          Champion
        </Text>,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Champion",
          "disabled": false,
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Champion",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
